.github/workflows/release.yml

```diff
@@ -0,0 +1,88 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "*"
+
+jobs:
+  trigger-buildkite-pipeline:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Trigger a Buildkite Build
+        uses: "buildkite/trigger-pipeline-action@v2.0.0"
+        with:
+          buildkite_api_access_token: ${{ secrets.TRIGGER_BK_BUILD_TOKEN }}
+          pipeline: "anza/agave-secondary"
+          branch: "${{ github.ref_name }}"
+          commit: "HEAD"
+          message: ":github: Triggered from a GitHub Action"
+
+  draft-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Create Release
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.rest.repos.createRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag_name: '${{ github.ref_name }}',
+              name: 'Release ${{ github.ref_name }}',
+              body: '🚧',
+              draft: true,
+              prerelease: false
+            })
+
+  version-bump:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Parse Info
+        id: parse_info
+        run: |
+          # get the next version
+          version=${{ github.ref_name }}
+          major=$(echo $version | cut -d'.' -f1)
+          minor=$(echo $version | cut -d'.' -f2)
+          patch=$(echo $version | cut -d'.' -f3)
+          next_version=$major.$minor.$((patch+1))
+          : "${next_version:?}"
+
+          # get the traget branch
+          target_branch=$major.$minor
+          : "${target_branch:?}"
+
+          echo "next_version=$next_version" | tee -a $GITHUB_OUTPUT
+          echo "target_branch=$target_branch" | tee -a $GITHUB_OUTPUT
+
+      - name: Create branch and make changes
+        run: |
+          next_version=${{ steps.parse_info.outputs.next_version }}
+
+          git checkout -b version-bump-$next_version
+          ./scripts/increment-cargo-version.sh patch
```

Comment:
We'll need to modify the PR to revert the hashbrown and proc-macro2 version changes.

I think we can get rid of that problem by bumping the hashbrown version. I'll create an issue for that one today.